infer/environment.cc

```diff
@@ -437,16 +437,18 @@ void Environment::updateKnowledge(core::Context ctx, cfg::LocalRef local, core::
             }
             whoKnows.sanityCheck();
         }
-    } else if (send->fun == core::Names::eqeq() || send->fun == core::Names::neq()) {
+    } else if (send->fun == core::Names::eqeq() || send->fun == core::Names::equal_p() ||
+               send->fun == core::Names::neq()) {
         if (!knowledgeFilter.isNeeded(local)) {
             return;
         }
         auto &whoKnows = getKnowledge(local);
         const auto &argType = send->args[0].type;
         const auto &recvType = send->recv.type;
 
-        auto &truthy = send->fun == core::Names::eqeq() ? whoKnows.truthy : whoKnows.falsy;
-        auto &falsy = send->fun == core::Names::eqeq() ? whoKnows.falsy : whoKnows.truthy;
+        bool funIsEq = send->fun == core::Names::eqeq() || send->fun == core::Names::equal_p();
```

Comment:
You could also do something like
```cpp
auto &truthy = send->fun == core::Names::neq() ? whoKnows.falsy : whoKnows.truthy;
```
but I thought the extra line was preferable to the double negative.